api/envoy/extensions/filters/http/ratelimit/v3/rate_limit.proto

```diff
@@ -119,4 +363,8 @@ message RateLimitPerRoute {
 
   // Specifies if the rate limit filter should include the virtual host rate limits.
   VhRateLimitsOptions vh_rate_limits = 1 [(validate.rules).enum = {defined_only: true}];
```

Comment:
Technically we just need to rename this enum, right? The behavior stays the same. More specific config can either override less specific config if it is present, include (merge), or ignore (which works like override but if config is not present it effectively disables less specific config).

Do you think we need to create new enum for this? Or can we just rename the existing enum?

Reply:
Renaming the existing enum would be a breaking change.  The options are to either create a new enum and deprecate the existing one or to just live with the current name.  Either way, though, the comments here need to be expanded to explain that this mechanism is not actually VirtualHost-specific.

Reply:
Makes sense. I think for clarity we should copy, rename this enum (to something like 'LessSpecificConfigBehavior' marking it hidden in this PR. Deprecation will be handled as part of implementing it in Envoy.

Reply:
Actually, I already went ahead to create a new enum and handle the deprecation and conversion between new and old field since I think keeping the old name is not ideal and accurate.

I have the local changes working and will send out for you guys review soon after some code clean up.

Reply:
I added a new enum `RateLimitsOptions` with a `DEFAULT` field which is used to differentiate the new config and old config. (line#277 in ratelimit.cc)